src/lib/rest/rest.cpp

```diff
@@ -262,18 +262,236 @@ static int uriArgumentGet(void* cbDataP, MHD_ValueKind kind, const char* ckey, c
+{
+  HttpHeaders*       headerP        = &ciP->httpHeaders;
+  char*              rest           = NULL;
+  char*              cP             = (char*) value;
+  HttpAcceptHeader*  acceptHeaderP;
+  char*              delimiter;
+
+  if (value[0] == 0)
+  {
+    ciP->httpStatusCode    = SccBadRequest;
+    ciP->acceptHeaderError = "empty item in accept header";
+    ciP->outMimeType       = mimeTypeSelect(ciP, "Error");
+    return false;
+  }
+
+  if ((delimiter = strchr(cP, ';')) != NULL)
+  {
+    *delimiter = 0;
+    rest = &delimiter[1];
+  }
+
+  //
+  // Now we have the 'media-range'.
+  // The broker accepts only the following two media types:
+  //   - application/json
+  //   - text/plain
+  // So, if the media-range is anything else, it is rejected immediately and not put in the list
+  //
+  if ((strcmp(cP, "*/*")              != 0) &&
+      (strcmp(cP, "application/*")    != 0) && 
+      (strcmp(cP, "application/json") != 0) && 
+      (strcmp(cP, "text/*")           != 0) &&
+      (strcmp(cP, "text/plain")       != 0))
+  {
+    return true;  // No error, just a media type that the broker doesn't recognize
+  }
+
+
+  acceptHeaderP = new HttpAcceptHeader();
+  acceptHeaderP->mediaRange = cP;
+  acceptHeaderP->qvalue     = 1;  // default value of 'q' - may be modified later
+
+  // If nothing after the media-range, we are done
+  if (rest == NULL)
+  {
+    headerP->acceptHeaderV.push_back(acceptHeaderP);
+    return true;
+  }
+
+  // If we get here, next in line must be a 'q', perhaps preceded by whitespace
+  while ((*rest == ' ') || (*rest == '\t'))
+  {
+    ++rest;
+  }
+
+  if (*rest == 0)
+  {
+    headerP->acceptHeaderV.push_back(acceptHeaderP);
+    return true;
+  }
+
+
+  //
+  // Next item is q=qvalue
+  //
+
+  if (*rest != 'q')
+  {
+    ciP->acceptHeaderError = "q missing in accept header";
+    ciP->httpStatusCode    = SccBadRequest;
+    delete acceptHeaderP;
+    return false;
+  }
+
+  // Pass 'q' and check for '='
+  ++rest;
+  if (*rest != '=')
+  {
+    ciP->acceptHeaderError = "missing equal-sign after q in accept header";
+    ciP->httpStatusCode    = SccBadRequest;
+    delete acceptHeaderP;
+    return false;
+  }
+  
+  // Pass '=' and check for Number
+  ++rest;
+  // Zero-out ';' if present
+  if ((cP = strchr(rest, ';')) != NULL)
+  {
+    *cP = 0;
+  }
+
+  // qvalue there?
+  if (*rest == 0)
+  {
+    ciP->acceptHeaderError = "qvalue in accept header is missing";
+    ciP->httpStatusCode    = SccBadRequest;
+    delete acceptHeaderP;
+    return false;
+  }
+
+  // qvalue a number?
+  if (str2double(rest, &acceptHeaderP->qvalue) == false)
+  {
+    ciP->acceptHeaderError = "qvalue in accept header is not a number";
+    ciP->httpStatusCode    = SccBadRequest;
+    ciP->outMimeType       = mimeTypeSelect(ciP, "Error");
+    delete acceptHeaderP;
+    return false;
+  }
+
+
+  //
+  // And now the accept-extensions  (which is ignored for now)
+  // FIXME P4: Implement treatment of accept-extensions
```

Comment:
The comment header of HttpAcceptHeader in HttpHeaders.h - that's where the whole thing is explained.

NTC